filebeat/beater/filebeat.go

```diff
@@ -22,7 +22,9 @@ type Filebeat struct {
 
 // New creates a new Filebeat pointer instance.
 func New() *Filebeat {
-	return &Filebeat{}
+	return &Filebeat{
+		config: &cfg.DefaultConfig,
```

Comment:
don't take the default config as pointer. DefaultConfig normally serves as template to be copied and must not contain any pointers!

Rather allocate new config initialized with DefaultConfig like:

```
config := cfg.DefaultConfig
return &Filebeat{ config: &config }
```